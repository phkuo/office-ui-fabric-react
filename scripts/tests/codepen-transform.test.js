const transform = require('../tasks/codepen-examples-transform');
const jscodeshift = require('jscodeshift');
const fs = require('fs');

describe('Codepen Transform', () => {
  it('correctly transforms code examples with variable exports', () => {
    const filePath = '../packages/office-ui-fabric-react/src/components/Label/examples/Label.Basic.Example.tsx';
    const fileInfo = {
      path: filePath,
      source: fs.readFileSync(filePath)
    };
    const api = { jscodeshift: jscodeshift.withParser('babylon'), stats: {} };
    const transformResult = transform(fileInfo, api);
    expect(transformResult).toMatchSnapshot();
  });

  it('correctly transforms code examples with class exports', () => {
    const filePath = '../packages/office-ui-fabric-react/src/components/TextField/examples/TextField.Basic.Example.tsx';
    const fileInfo = {
      path: filePath,
      source: fs.readFileSync(filePath)
    };
    const api = { jscodeshift: jscodeshift.withParser('babylon'), stats: {} };
    const transformResult = transform(fileInfo, api);
    expect(transformResult).toMatchSnapshot();
  });
});
