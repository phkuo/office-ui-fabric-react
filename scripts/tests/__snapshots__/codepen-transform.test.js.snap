// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Codepen Transform correctly transforms code examples with class exports 1`] = `
"let { TextField, MaskedTextField, Fabric } = window.Fabric;

class TextFieldBasicExample extends React.Component<any, any> {
  public render(): JSX.Element {
    return (
      <div className=\\"docs-TextFieldExample\\">
        <TextField label=\\"Standard\\" />
        <TextField label=\\"Disabled\\" disabled={true} />
        <TextField label=\\"Read Only\\" readOnly={true} />
        <TextField label=\\"Required \\" required={true} />
        <TextField label=\\"With error message\\" errorMessage=\\"Error message\\" />
        <MaskedTextField label=\\"With input mask\\" mask=\\"m\\\\ask: (999) 999 - 9999\\" />
      </div>
    );
  }
}

ReactDOM.render(<TextFieldBasicExample />, document.getElementById('content'));
"
`;

exports[`Codepen Transform correctly transforms code examples with variable exports 1`] = `
"let { Label, Fabric } = window.Fabric;

ReactDOM.render(<LabelBasicExample />, document.getElementById('content'));
"
`;
